drv/transceivers-server/src/net_api.rs

```diff
@@ -0,0 +1,410 @@
+        }),
+        1 => Ok(FpgaPortMasks {
+            left: 0,
+            right: fpga_ports,
+        }),
+        i => Err(Error::InvalidFpga(i)),
+    }
+}
+
+impl ServerImpl {
+    /// Attempt to read and handle data from the `net` socket
+    pub fn check_net(
+        &mut self,
+        rx_data_buf: &mut [u8],
+        tx_data_buf: &mut [u8],
+    ) {
+        const SOCKET: SocketName = SocketName::transceivers;
+
+        match self.net.recv_packet(
+            SOCKET,
+            LargePayloadBehavior::Discard,
+            rx_data_buf,
+        ) {
+            Ok(mut meta) => {
+                // Modify meta.size based on the output packet size
+                meta.size = match hubpack::deserialize(rx_data_buf) {
+                    Ok((msg, data)) => {
+                        let (reply, data_len) = match self.handle_message(
+                            msg,
+                            data,
+                            &mut tx_data_buf[Message::MAX_SIZE..],
+                        ) {
+                            Ok(r) => r,
+                            Err(e) => (HostResponse::Error(e), 0),
+                        };
+                        let out = Message {
+                            header: msg.header,
+                            modules: msg.modules,
+                            body: MessageBody::HostResponse(reply),
+                        };
+                        // Serialize into the front of the tx buffer
+                        let msg_len =
+                            hubpack::serialize(tx_data_buf, &out).unwrap();
+
+                        // At this point, any supplementary data is written to
+                        // tx_buf[Message::MAX_SIZE..].  Let's shift it
+                        // backwards based on the size of the leading `Message`:
+                        tx_data_buf.copy_within(
+                            Message::MAX_SIZE..(Message::MAX_SIZE + data_len),
+                            msg_len,
+                        );
+                        (msg_len + data_len) as u32
+                    }
+                    Err(_e) => hubpack::serialize(
+                        tx_data_buf,
+                        &Message {
+                            header: Header {
+                                version: 1,
+                                message_id: u64::MAX,
+                            },
+                            modules: ModuleId::default(),
+                            body: MessageBody::HostResponse(
+                                HostResponse::Error(Error::ProtocolError),
+                            ),
+                        },
+                    )
+                    .unwrap() as u32,
+                };
+
+                self.net
+                    .send_packet(
+                        SOCKET,
+                        meta,
+                        &tx_data_buf[..meta.size as usize],
+                    )
+                    .unwrap();
```

Comment:
Is this temporary / part of the draft? Handling `QueueFull` in particular might be slightly annoying, so maybe worth doing sooner rather than later.

Reply:
Fixed to explicitly drop packets on `QueueFull` (still panicking on other errors, since they're not expected)